### PR TITLE
Adds FAQ on ParentSpanId

### PR DESCRIPTION
**Pull request by codefromthecrypt**

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
cc @openzipkin/core @openzipkin/instrumentation-owners This is an overdue section, which underpins a frequently asked question.

A follow-up will explain the single-host-span model which tosses this field.

**Review by shakuzen — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
current status: so far B3 doesn't discuss that traces are a graph explicitly. It might be helpful to explain that parent-child is required to model the original call graph. then we wouldn't have to explain here the same thing.

**Comment by codefromthecrypt** [comment]
Removed nuance about storage impact of attempting to go without a parent, as it distracted from the more general point.

Here's the nuance discussion just as a log:

>The parent span id is propagated to eliminate ambiguity about if a span is a root or not. It also removes a race condition creating the span (since the server often completes before its client processes the response).
>
>If the server reported the span without its parent, the tracing system would likely assume it is a root span (a root is a span with no parent). The tracing pipeline could delay write until it is known whether the span had a client or not. However such an approach would imply consistent routing on trace ID and also state to keep these mappings. The simpler approach is to propagate the ParentSpanId when present. This removes the race condition and confusion around if a span is a root or not.

**Comment by codefromthecrypt** [comment]
@jcarres-mdsol @shakuzen PTAL I think it is more accessible now, but check me

File: README.md [hunk 1]
```diff
@@ -92,3 +92,39 @@ explicitly or implicitly down-cased.
...
 For example, the Http header `X-B3-ParentSpanId: 0020000000000001` would become
 an ASCII header `x-b3-parentspanid` with the same value.
+
+# Frequently Asked Questions
+
+## Why is ParentSpanId propagated?
+
+In B3, the trace context is extracted from headers sent upstream. Timing and
+metadata for the client and server side of an operation are recorded against
+the same context. So, the `ParentSpanId` is the ID of the operation that caused
```
Comment (jcarres-mdsol) [inline README.md]: Probably does not need `So,`

File: README.md [hunk 2]
```diff
@@ -92,3 +92,39 @@ explicitly or implicitly down-cased.
...
+
+```
+                              ┌───────────────────┐
+ Incoming Request Headers     │   TraceContext    │
+┌───────────────────┐         │ ┌───────────────┐ │
+│ XXXX─TraceId      │─────────┼─┼> TraceId      │ │
+│                   │extract()│ │               │ │
+│ XXXX─SpanId       │─────────┼─┼> ParentSpanId │ │
+└───────────────────┘         │ │               │ │      ┌──────────────┐
+                              │ │  SpanId      <┼─next()─│ ID Generator │
```
Comment (jcarres-mdsol) [inline README.md]: First box is a service (probably) , second box is a service but third box is functionality inside a library. That makes it confusing

File: README.md [hunk 3]
```diff
@@ -92,3 +92,39 @@ explicitly or implicitly down-cased.
...
+
+In B3, the trace context is extracted from headers sent upstream. Timing and
+metadata for the client and server side of an operation are recorded against
+the same context. So, the `ParentSpanId` is the ID of the operation that caused
+the current RPC. For example, it could be the ID of another server request
+or a scheduled job. `ParentSpanId` is propagated so that when data is reported
+to Zipkin, it can be placed in the correct spot in the trace tree.
+
+Some propagation formats look similar to B3, but don't propagate a field named
+parent. People with these formats may wonder why B3 has `X-B3-ParentSpanId` and
```
Comment (jcarres-mdsol) [inline README.md]: I would skip the sentence starting by `People...`  and just say: `Usually these formats propagate a span ID field which serves the same purpose as ParentSpanId`

File: README.md [hunk 4]
```diff
@@ -92,3 +92,39 @@ explicitly or implicitly down-cased.
 
 For example, the Http header `X-B3-ParentSpanId: 0020000000000001` would become
 an ASCII header `x-b3-parentspanid` with the same value.
+
+# Frequently Asked Questions
+
+## Why is ParentSpanId propagated?
+
+In B3, the trace context is extracted from headers sent upstream. Timing and
```
Comment (jcarres-mdsol) [inline README.md]: Maybe
`In B3, the trace context is extracted from headers (the SpanID)` 
I really do not know what `sent upstream` means

**Comment by codefromthecrypt** [comment]
>
> +```
> +                              ┌───────────────────┐
> + Incoming Request Headers     │   TraceContext    │
> +┌───────────────────┐         │ ┌───────────────┐ │
> +│ XXXX─TraceId      │─────────┼─┼> TraceId      │ │
> +│                   │extract()│ │               │ │
> +│ XXXX─SpanId       │─────────┼─┼> ParentSpanId │ │
> +└───────────────────┘         │ │               │ │      ┌──────────────┐
> +                              │ │  SpanId      <┼─next()─│ ID Generator │
>
> First box is a service (probably) , second box is a service but third box
> is functionality inside a library. That makes it confusing
>
no the first box is headers the second box is a trace context and the last
is an ID generator. they are all things the tracer would be acting on.
ex in ruby, they would be...

env[:request_headers]
Trace::TraceId
TraceGenerator

File: README.md (hunk 4 above)
Comment (adriancole) [inline README.md]: SpanId is not universally used, which is why I'm avoiding that.

Comment (codefromthecrypt) [reply README.md]: TraceContext is used further up in the doc.. replacing "sent upstream" with incoming (which is more often used. thx)

**Comment by codefromthecrypt** [comment]
clarified diagram as a library diagram (as opposed to a service one). added a diagram to show the difference. pruned words a bit taking suggestions from @jcarres-mdsol 

third time's a charm? :)

**Comment by codefromthecrypt** [comment]
since this is editing text, we can do more edits later if needed. thanks for the reviews!

**Pull request merged**